api/v1beta1/temporalappworker_types.go

```diff
@@ -0,0 +1,101 @@
+// Licensed to Alexandre VILAIN under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Alexandre VILAIN licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v1beta1
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TemporalAppWorkerSpec defines the desired state of TemporalAppWorker
+type TemporalAppWorkerSpec struct {
+	// Version defines the app worker version.
+	Version string `json:"version"`
+	// Image defines the temporal ui docker image the instance should run.
+	Image string `json:"image"`
+	// Number of desired replicas. Default to 1.
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Replicas *int32 `json:"replicas"`
+	// An optional list of references to secrets in the same namespace
+	// to use for pulling temporal images from registries.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	// TemporalServer connection details.
+	TemporalConnection *TemporalConnectionSpec `json:"temporalConnection"`
```

Comment:
I think it could be more "Kubernetes-way" to define a reference to the cluster ?
As it's done for `TemporalNamespace`: https://github.com/alexandrevilain/temporal-operator/blob/main/api/v1beta1/temporalnamespace_types.go#L28

Reply:
@alexandrevilain I need some help with this one. I did not see or understand how to store the TemporalConnectionSpec under ClusterRef so that I could retrieve the values set in the spec.
https://github.com/alexandrevilain/temporal-operator/blob/6026312960f2ec3d4d66c36ad000329df9361a5c/pkg/resource/appworker_deployment_builder.go#L75

Reply:
From the `clusterRef`, you can retrieve the targeted `TemporalCluster` object has done here:
https://github.com/alexandrevilain/temporal-operator/blob/main/controllers/temporalnamespace_controller.go#L69

Then using the `TemporalCluster` object you can get the frontend endpoint using `cluster.GetPublicClientAddress()`

Reply:
@alexandrevilain I tried something like this 
```
	namespacedName := types.NamespacedName{Namespace: b.instance.Namespace, Name: b.instance.Spec.ClusterRef.Name}
	cluster := &v1beta1.TemporalCluster{}
	err := client.Client.Get(context.TODO(), namespacedName, cluster)
	if err != nil {
		return fmt.Errorf("failed getting cluster ref: %v", err)
	}
```
But it isnt working, not sure if the client is usable in the deployment builders without passing it in?

Reply:
Nope, you have to get the cluster from the reconciliation func, as done here: https://github.com/alexandrevilain/temporal-operator/blob/main/controllers/temporalnamespace_controller.go#L69 :)

Reply:
Yes, I understand :) 
I think that the reconcilier should provide both `TemporalWorkerProcess` and `TemporalCluster` to the builder instance. Then you can get the frontend endpoint. 
I hope it's clearer, if not let me know I will write an example

Reply:
@alexandrevilain yeah i have played around with passing client through to builders via reconciler, I couldnt get this to work so I am missing something. If you can get TemporalCluster and TemporalWorkerProcess passed in to builder that would be great :)